tests/test_lib_audit.py

```diff
@@ -193,3 +194,31 @@ def test_06_truncate_data(self):
                          column_length.get("serial"))
         self.assertEqual(len(self.Audit.audit_data.get("token_type")),
                          column_length.get("token_type"))
+
+    def test_07_sign_non_ascii_entry(self):
+        # Log a username as unicode with a non-ascii character
+        self.Audit.log({"serial": "1234",
+                        "action": "token/assign",
+                        "success": True,
+                        "user": u"kölbel"})
+        self.Audit.finalize_log()
+        audit_log = self.Audit.search({"user": u"kölbel"})
+        self.assertEqual(audit_log.total, 1)
+        self.assertEqual(audit_log.auditdata[0].get("user"), u"kölbel")
+        self.assertEqual(audit_log.auditdata[0].get("sig_check"), "OK")
+
+        # Log a username as utf-8 with a non-ascii character
+        username = u"kölbel".encode("utf-8")
+        # hm, even if we should log a utf8 username...
+        self.Audit.log({"serial": "1234",
+                        "action": "token/assign",
+                        "success": True,
+                        "user": username})
```

Comment:
Hmm, looking into this, I think this doesn't actually work -- shouldn't the query below give two results?

The log says:
```
[2017-07-21 14:45:05,539][3049][140085362857024][ERROR][privacyidea.lib.auditmodules.sqlaudit:238] exception ProgrammingError('(sqlite3.ProgrammingError) You must not use 8-bit bytestrings unless you use a text_factory that can interpret 8-bit bytestrings (like text_factory = str). It is highly recommended that you instead just switch your application to Unicode strings.',)
[2017-07-21 14:45:05,540][3049][140085362857024][ERROR][privacyidea.lib.auditmodules.sqlaudit:239] DATA: {'action': 'token/assign', 'serial': '1234', 'user': 'k\xc3\xb6lbel', 'success': True}
```